File: backend/src/lakehouse/schemas/embeddings.py
from pydantic import BaseModel, Field


class Embedding3DPoint(BaseModel):
    chunk_key: str = Field(...)
    x: float = Field(...)
    y: float = Field(...)
    z: float = Field(...)
    conference_date: str = Field(...)


class Embedding3DResponse(BaseModel):
    points: list[Embedding3DPoint] = Field(default_factory=list)

File: frontend/src/api/embeddings.ts
import { apiClient } from './client'

export interface Embedding3DPoint {
  chunk_key: string
  x: number
  y: number
  z: number
  conference_date: string
  cluster_id?: number
}

export interface Embedding3DResponse {
  points: Embedding3DPoint[]
}

export async function getEmbeddings3D(): Promise<Embedding3DResponse> {
  return apiClient<Embedding3DResponse>('/embeddings/3d')
}
